### ISSUE TITLE
Feature Request: Use the cli with --public and a fixed seed

### ISSUE THREAD

**Issue by oren-z0** [holesail/holesail#29]
Running `holesail --public` is great but if the process restarts it creates a new connection-string every time.
I want it to create a fixed connection that I can give to friends without having to send them updates after reboots.

**Comment by supersuryaansh** [comment]
@oren-z0 Yeah it is not a very good user design, this will be fixed in our upcoming updates.

**Comment by jjacke13** [comment]
if i may propose something, @oren-z0  you can try something like: holesail --live 80 --connector $(cat remote.txt) , assuming you have saved the connection string to a file called remote.txt. I use this way in Nixtcloud ( https://github.com/jjacke13/nixtcloud/blob/main/Rpi4/configuration.nix --> lines 109-116 & 145-155). Plus you can also configure it to create a new string if you delete the file remote.txt (this in case you have private strings).

**Comment by oren-z0** [comment]
@jjacke13 This is the symmetric key, meaning the other side can use it to pretend to be a server.
If you are the manager of both sides this is fine, but if you want to open a website on your local machine to other remote users (i.e. that use [holesail-go](https://play.google.com/store/apps/details?id=io.holesail.holesail.go)), it is problematic.

**Comment by jjacke13** [comment]
If I understand correctly, you want to share your public connection string with others, so they can connect to your site. So you have to give them the connection string. So they will know the connection string and they can do whatever they want with it (regardless of where you decide to store it on your server).

**Comment by supersuryaansh** [comment]
@jjacke13 They want to start a public connection but have the same connection string every time, which is not supported at the moment na dis a planned feature.

**Comment by oren-z0** [comment]
@supersuryaansh I couldn't wait: https://github.com/holesail/holesail/pull/30

@jjacke13 I want to share the connection string with third parties, but I don't want them to be able to create a server that will take over connection attempts to my server. And as @supersuryaansh said, I want it to be the same connection string every time, i.e. generated randomly once (by me), and not re-generated on every execution.

**Comment by jjacke13** [comment]
Maybe I get something wrong. Just create a random string in advance, save it somewhere and then call holesail with that string as --connector.

**Comment by jjacke13** [comment]
> @supersuryaansh I couldn't wait: https://github.com/holesail/holesail/pull/30
> 
> @jjacke13 I want to share the connection string with third parties, but I don't want them to be able to create a server that will take over connection attempts to my server. And as @supersuryaansh said, I want it to be the same connection string every time, i.e. generated randomly once (by me), and not re-generated on every execution.

But aren't public connectors used with that characteristic in mind?

**Comment by oren-z0** [comment]
@jjacke13 Yes, public connectors are used like that, but the current cli does not allow to use `--public` and `--connector` together. If you use `--public` alone it will generate a new private-public key pair, showing you the public connector and keeping the private key internally. If the server is rebooted, it will generate a new key pair, and the third party users will not be able to connect to the old connector. So instead of generating a random key pair on every boot, I want to use a given seed (32 bytes that I generated randomly once and saved in a file).

**Comment by jjacke13** [comment]
> @jjacke13 Yes, public connectors are used like that, but the current cli does not allow to use `--public` and `--connector` together. If you use `--public` alone it will generate a new private-public key pair, showing you the public connector and keeping the private key internally. If the server is rebooted, it will generate a new key pair, and the third party users will not be able to connect to the old connector. So instead of generating a random key pair on every boot, I want to use a given constant seed (32 bytes that I generated randomly once and saved in a file).

Ohh that is what I missed. --public and --connector together. I hadn't realized that you can't use them together

**Comment by supersuryaansh** [comment]
Solved in V2.
